### PR TITLE
EO-9856

### PR DESCRIPTION
**Pull request by hjaltisan**
EO-9856
 - Added the following for log functions to the console (so that the scripts can call them): console.info(), console.warning(), console.critical().
 - Added some minor log lines here and there.
 - Added log lines for the result of the http call (although the result does not tell us which post it belongs to.
 - Added automatic logging to the installer, that creates a log file on each run and stores in the same location as the launcher logs. It also follows the naming convention used there. More on the wiki.
 - Added the option to log to a file named install.log that will be created in the same directory as the installer. This is turned on by providing -f or --log-to-file to the installer on startup. This works in tandem with -v/--verbose.
 - Added the option to start the installer with automatic logging disabled (--no-automatic-logging)
 - Disabled the InstallationLog.txt from being created in the install directory upon successful installation. We could add a startup parameter to enable it, but I don't see any need for it.

### CODE REVIEW

**Review by BaldvinTh — APPROVED** [review]
Looks good to me :)

File: src/libs/installer/httpthreadworker.cpp [hunk 1]
```diff
@@ -27,11 +28,20 @@ void HttpThreadWorker::process(const QByteArray& data, const QString& url)
 
     QNetworkReply* reply = m_network->post(*request, data);
 
-    connect(reply, &QNetworkReply::finished, [this, reply]() { m_finished++; reply->deleteLater();} );
+    connect(reply, &QNetworkReply::finished, [this, reply]() {
+        QVariant status_code = reply->attribute(QNetworkRequest::HttpStatusCodeAttribute);
+        QString status = status_code.toString();
+        qDebug() << "framework | HttpThreadWorker::process | status =" << status;
+        m_finished++; reply->deleteLater();
```
Comment (rosabj) [inline src/libs/installer/httpthreadworker.cpp:36]: I'd put these in separate lines. Same below.

File: src/libs/installer/init.cpp [hunk 2]
```diff
@@ -104,6 +105,47 @@ static QString trimAndPrepend(QtMsgType type, const QString &msg)
     return ba;
 }
 
+static QString logFiletrimAndPrepend(QtMsgType type, const QString &msg)
+{
+    QString ba(msg);
+    // last character is a space from qDebug
+    if (ba.endsWith(QLatin1Char(' ')))
```
Comment (rosabj) [inline src/libs/installer/init.cpp:112]: You could replace this with QString::trimmed() It does remove whitespace from the start as well, but I can't see that that matters here. QString::simplified() could also work. That does the same as trimmed, but also replaces internal whitespace sequences with a single space.

File: src/libs/installer/init.cpp [hunk 3]
```diff
@@ -104,6 +105,47 @@ static QString trimAndPrepend(QtMsgType type, const QString &msg)
...
+        ba.chop(1);
+
+    // remove quotes if the whole message is surrounded with them
+    if (ba.startsWith(QLatin1Char('"')) && ba.endsWith(QLatin1Char('"')))
+        ba = ba.mid(1, ba.length() - 2);
```
Comment (rosabj) [inline src/libs/installer/init.cpp:117]: It seems a bit off to have to do this here. Aren't these QDebug messages? QDebug::noquote() could be used when creating them. Those that are just a single QChar, QString or QByteAray that is.

File: src/libs/installer/installereventoperation.cpp [hunk 4]
```diff
@@ -254,6 +269,7 @@ bool InstallerEventOperation::sendInstallerEvent(QStringList args)
     }
     else if (event == QString::fromLatin1("ShutDown") && args.size() == 5)
     {
+        qDebug() << "framework | InstallerEventOperation::sendInstallerEvent | ShutDown | args =" << args;
```
Comment (rosabj) [inline src/libs/installer/installereventoperation.cpp:272]: I'm on the fence, but this is starting to feel a bit repetitive. Like maybe there could be a function for this that always at least prepends framework, accepts the function name and puts that in its place...?

File: src/libs/installer/init.cpp (hunk 2 above)
Comment (hjaltisan) [reply src/libs/installer/init.cpp:112]: This code is a copy paste from `trimAndPrepend` above (that was written by Qt, so I didn't really think about it. But of course there is nothing stopping us from changing it anyways.
But if we're going to change this, then we should do it in their code as well.

File: src/libs/installer/init.cpp (hunk 3 above)
Comment (hjaltisan) [reply src/libs/installer/init.cpp:117]: This code is a copy paste from `trimAndPrepend` above (that was written by Qt, so I didn't really think about it. But of course there is nothing stopping us from changing it anyways.
But if we're going to change this, then we should do it in their code as well.

File: src/libs/installer/installereventoperation.cpp (hunk 4 above)
Comment (hjaltisan) [reply src/libs/installer/installereventoperation.cpp:272]: A bit repetitive? You mean a lot 😃 
I didn't have the time to put that in there. I'll add it to little things:
 * We need 5 new logger functions to be able to prepend `framework | ` to all our logs.
 * We need to replace our q<MESSAGELEVEL> calls with these new debug functions

File: src/libs/installer/httpthreadworker.cpp (hunk 1 above)
Comment (hjaltisan) [reply src/libs/installer/httpthreadworker.cpp:36]: Done

**Pull request merged**